pubsub/gossipsub/gossipsub-v1.4.md

```diff
@@ -0,0 +1,164 @@
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+# Overview
+
+This document outlines small extensions to the [gossipsub 
+v1.2](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.2.md) 
+protocol to improve performance when handling large message transmissions. 
+The extensions are optional, fully compatible with v1.2 of the protocol, 
+and involve minor modifications to peer behavior when receiving or forwarding large messages.  
+
+The proposed modifications address the issue that the number of IWANT requests and duplicates increases significantly with the message size. 
+This happens because sending a large message can take considerable time, 
+and during this time, the receiver is unaware of the IDs of the messages it is receiving.
+
+Under the current arrangement, if IHAVE announcements are received for a message that is already being received, 
+the receiver may generate multiple IWANT requests, triggering unnecessary retransmissions of the same message. 
+Higher message reception time also increases the probability of simultaneously receiving the same message from many peers. 
+
+Prepending a preamble while sending a large message can allow receivers 
+to immediately gain insights into the IDs and lengths of the messages they are receiving.  
+This can allow receivers to defer IWANT requests for messages they are currently receiving, 
+thereby reducing the number of unnecessary IWANT requests.
+
+At the same time, receivers may inform their mesh members about ongoing message reception.
+For this purpose, a new control message (IMReceiving) is introduced, 
+indicating that the peer sending IMReceiving message is currently receiving messages identified by their message IDs, 
+and sending duplicates might be unnecessary.
+
+On receiving an IMReceiving message, a peer should defer sending of messages identified by the message IDs in IMReceiving message.
+This can lead to a significant reduction in bandwidth utilization and message latency.  
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.4.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.4.0`
+
+## Parameters
+
+This section lists the configuration parameters that clients need to agree on to avoid peer penalizations.
+
+| Parameter                | Description                                                      | Reasonable Default |
+|--------------------------|------------------------------------------------------------------|--------------|
+| `peer_preamble_announcements` | The maximum number of preamble announcements for unfinished transfers per peer | 1???  |
+| `mesh_preamble_announcements` | The maximum number of preamble announcements to accept for unfinished transfers per topic per heartbeat interval | 3???  |

```

Comment:
Do you think we need separate parameters for these limits? I would treat skipping a message body or supplying a message of a different length as a regular protocol violation (tracked with `behaviourPenaltyWeight`) That should  downscore a bad peer pretty quickly.